schema/defs-image.json

```diff
@@ -4,7 +4,7 @@
     "mediaType": {
       "id": "https://opencontainers.org/schema/image/mediaType",
       "type": "string",
-      "pattern": "^[a-z]+/[0-9a-zA-Z.+]+$"
+      "pattern": "^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}/[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}$"
```

Comment:
That change is cherry-picked from #448, which focuses on updating the `mediaType` pattern to match [RFC 6838][1].  And yeah, for some reason they allow both `#` and `^` :p.

I don't think an enum would work, because descriptors are generic.  On the other hand, particular *instances* of descriptors may not be generic, and #404 is an example of using enums in those cases.

[1]: https://tools.ietf.org/html/rfc6838#section-4.2